auth/user.go

```diff
@@ -183,6 +183,98 @@ func (user *userImpl) RoleHistory() TimedSetHistory {
 	return user.RoleHistory_
 }
 
+func (user *userImpl) RevokedChannels(since uint64) map[string]uint64 {
+
+	// Grab all channels user has access to
+	accessibleChannels := user.InheritedChannels()
+
+	// Get revoked roles
+	rolesToRevoke := map[string]uint64{}
+	roleHistory := user.RoleHistory()
+	for roleName, history := range roleHistory {
+		if !user.RoleNames().Contains(roleName) {
+			for _, entry := range history.Entries {
+				if entry.StartSeq <= since && entry.EndSeq >= since {
```

Comment:
Having both StartSeq and EndSeq comparisons as "or equal to" doesn't look correct - my guess is that it should be simply `entry.EndSeq > since`, as a call with since=endSeq usually implies everything up to and including endSeq has already been replicated.  But unit tests with boundary conditions would be the best approach.

Reply:
Yeah, you're correct would end up with unnecessary revocation message. Confirmed with test.